client/consul_template_test.go

```diff
@@ -905,3 +911,125 @@ func TestTaskTemplateManager_Signal_Error(t *testing.T) {
 		t.Fatalf("Unexpected error: %v", harness.mockHooks.KillReason)
 	}
 }
+
+// TestTaskTemplateManager_Env asserts templates with the env flag set are read
+// into the task's environment.
+func TestTaskTemplateManager_Env(t *testing.T) {
+	template := &structs.Template{
+		EmbeddedTmpl: `
+# Comment lines are ok
+
+FOO=bar
+foo=123
+ANYTHING-goes=Spaces are=ok!
```

Comment:
Newlines need to be escaped. The 2 easiest ways are probably:

1. `{{ key "derp" | toJSON}}`
2. `{{ key "derp" | printf "%q" }}`

Both will properly escape newlines.

However, in this current PR they won't be unescaped and if your task read the environment it would look like:

```go
derp := os.Getenv("derp")
fmt.Printf(">>%s<<\n", derp)
```

It would print the _escaped_ value:
```
>>"hello\nworld"<<
```

So I'll post a followup PR that properly un-escapes values, so that the code above would print:

```
>>hello
world<<
```